test/e2e/helpers.go

```
@@ -0,0 +1,611 @@
+				return fmt.Sprintf("%s:%d", nodeIP, port.NodePort), nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("no suitable port found in gateway service %s/%s", gatewayNamespace, gatewayName)
+}
+
+// HTTPGet performs an HTTP GET request to a WasmModule URL.
+// For local Kind clusters, it routes through the ingress with Host header.
+// Retries on 404 errors to handle ingress propagation delays.
+func (tc *TestContext) HTTPGet(ctx context.Context, targetURL string) (string, error) {
```

golangci-lint (2.5.0)

[error] calculated cyclomatic complexity for function HTTPGet is 11, max is 10

(cyclop)